### PR TITLE
MAGENTO-2256 - append child_id to productIds if exist

### PR DESCRIPTION
**Pull request by awesselburg**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I updated the CHANGELOG.md

### CODE REVIEW

**Comment by augsteyer** [comment]
you got the new specs, I am ready for adjustments

File: extension/lib/magento/customer/getItems.js [hunk 1]
```diff
@@ -13,8 +13,13 @@ module.exports = async (context, input) => {
   const wishlistItems = await request.send(wishlistItemsEndpointUrl, 'Request to Magento: getFavorites', 'GET')
   context.storage.user.map.del('wishlistItemIdMapping')
   const productIds = wishlistItems.map(item => {
-    context.storage.user.map.setItem('wishlistItemIdMapping', item.product_id, item.wishlist_item_id)
-    return item.product_id
+    const { child_ids: childIds, type, product_id: productItemId } = item
```
Comment (augsteyer) [inline extension/lib/magento/customer/getItems.js]: Maybe just use this and delete the line below?
```suggestion
    let { child_ids: childIds, type, product_id: productId } = item
```

File: extension/lib/magento/customer/getItems.js [hunk 2]
```diff
@@ -13,8 +13,13 @@ module.exports = async (context, input) => {
...
+    let productId = productItemId
+    if (type === 'configurable' && childIds && childIds.length === 1) {
```
Comment (augsteyer) [inline extension/lib/magento/customer/getItems.js]: we technically have tests for this in the cloud plugin, I don't think we should be this defensive. That is, if magento endpoint does return multiple id's for a configurable product OR no child ids, this implementation would only hide the problem. Just trust the endpoint imho. @Rekhyt what ya think?

**Review by augsteyer — CHANGES_REQUESTED** [review]

File: extension/lib/magento/customer/getItems.js [hunk 3]
```diff
@@ -13,8 +13,12 @@ module.exports = async (context, input) => {
   const wishlistItems = await request.send(wishlistItemsEndpointUrl, 'Request to Magento: getFavorites', 'GET')
   context.storage.user.map.del('wishlistItemIdMapping')
   const productIds = wishlistItems.map(item => {
-    context.storage.user.map.setItem('wishlistItemIdMapping', item.product_id, item.wishlist_item_id)
-    return item.product_id
+    let { child_ids: childIds, type, product_id: productId } = item
+    if (type === 'configurable') {
+      productId = `${productId}-${childIds[0]}`
```
Comment (augsteyer) [inline extension/lib/magento/customer/getItems.js]: I just thought of something, isn't it just this
```suggestion
      productId += `-${childIds[0]}`
```

File: extension/lib/magento/customer/getItems.js [hunk 4]
```diff
@@ -13,8 +13,12 @@ module.exports = async (context, input) => {
...
+    }
+    context.storage.user.map.setItem('wishlistItemIdMapping', productId, item.wishlist_item_id)
+    return productId
```
Comment (augsteyer) [inline extension/lib/magento/customer/getItems.js:22]: add a space above return

**Review by augsteyer — CHANGES_REQUESTED** [review]

**Comment by augsteyer** [comment]
@andre-kraus can you help me run the newman tests for this, my `sgconnect` keeps crashing

**Comment by augsteyer** [comment]
@awesselburg did you run the whole newman tests suite? I am having issues running the tests, but I can see that you added an extra product. You can follow the names of the tests to realize that this will cause a problem with the last test in the series.

**Comment by augsteyer** [comment]
what I mean is, double check that you can delete the `402-231` as well. Although, does the frontend try to delete `402` or `402-231`?

File: extension/lib/magento/customer/getItems.js (hunk 4 above)
Comment (andre-kraus) [reply extension/lib/magento/customer/getItems.js:22]: Done

File: extension/lib/magento/customer/getItems.js (hunk 2 above)
Comment (andre-kraus) [reply extension/lib/magento/customer/getItems.js]: Done

**Comment by augsteyer** [comment]
may need to fix the travis error and we are good to go

**Comment by andre-kraus** [comment]
> may need to fix the travis error and we are good to go

Seems like I can not. It's an npm audit issue -> dependency of newman "hawk" requires cryptiles "3.x.x" and has not been updated for 9 month. We need cryptiles 4.1.2 to fix the audit issue. ideas?

**Review by augsteyer — APPROVED** [review]

**Pull request merged**